netbox: fix `net.box` stream object `space` table `__index` metamethod (#8750)
tarantool/tarantool: changelogs/unreleased/gh-8598-netbox-stream-space-__index-bug.md

## bugfix/box

> * Fixed a bug that didn't allow accessing spaces named `_stream` and
>   `_stream_space_cache` and indexes named `_space` from `net.box` stream objects
>   (gh-8598).

Reviewer: ```suggestion
* Fixed a bug when it was not possible to access spaces named `_stream` and
  `_stream_space_cache` and indexes named `_space` using the `net.box` stream objects
  (gh-8598).
```

Author: Fixed.

> * Fixed a bug in the `net.box` stream object `space` table `__index` metamethod
>   that didn't allow accessing spaces with `_stream` and `_stream_space_cache`
>   names (gh-8598).

Reviewer: ```suggestion
* Fixed a bug in the `__index` metamethod of the `space` table:  
it was impossible to access spaces named `_stream` and `_stream_space_cache` 
using the `net.box` stream object (gh-8598).
```

Author: > Fixed a bug in the `__index` metamethod of the `space` table:

What `space` table are we talking about?
